object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/ConstantValueExpr.java

```diff
@@ -35,7 +35,7 @@ public String toString() {
 
       @Override
       public int compareTo(Object o) {
-         return 0;  //todo [anistor] hack!
+         return name.compareTo(((ParamPlaceholder) o).getName());
```

Comment:
The contract for compareTo allows failing with NullPointerException is arg is null